core/templates/pages/donate-page/donate-page.component.spec.ts

```diff
@@ -0,0 +1,104 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for donate page.
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { DonatePageComponent } from './donate-page.component';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { WindowDimensionsService } from
+  'services/contextual/window-dimensions.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+describe('Donate page', () => {
+  const siteAnalyticsServiceStub = new SiteAnalyticsService(
+    new WindowRef());
+
+  beforeEach(async() => {
+    TestBed.configureTestingModule({
+      declarations: [DonatePageComponent],
+      providers: [
+        {provide: SiteAnalyticsService, useValue: siteAnalyticsServiceStub},
+        UrlInterpolationService,
+        {
+          provide: WindowDimensionsService,
+          useValue: {
+            isWindowNarrow: () => true
+          }
+        },
+        {
+          provide: WindowRef,
+          useValue: {
+            nativeWindow: {
+              location: {
+                href: ''
+              }
+            }
+          }
+        }
+      ]
+    }).overrideComponent(DonatePageComponent, {
+      set: {
+        // The template is being overwritten because require is being
+        // used on component, which will throw an error on unit tests.
+        template: ''
+      }
+    }).compileComponents();
+  });
+  beforeEach(() => {
+    this.fixture = TestBed.createComponent(DonatePageComponent);
+    this.component = this.fixture.componentInstance;
+  });
+
+  it('should successfully instantiate the component', () => {
+    expect(this.component).toBeDefined();
+  });
+
+  it('should set component properties when ngOnInit() is called', () => {
+    this.component.ngOnInit();
+
+    expect(this.component.windowIsNarrow).toBe(true);
+    expect(this.component.donateImgUrl).toBe(
+      '/assets/images/general/opp_donate_text.svg');
+  });
+
+  it('should donate throught amazon sucessfully', (done) => {
+    spyOn(siteAnalyticsServiceStub, 'registerGoToDonationSiteEvent')
+      .and.callThrough();
+
+    this.component.onDonateThroughAmazon();
+    expect(siteAnalyticsServiceStub.registerGoToDonationSiteEvent)
+      .toHaveBeenCalledWith('Amazon');
+
+    setTimeout(() => {
+      expect(this.component.windowRef.nativeWindow.location.href).toBe(
+        'https://smile.amazon.com/ch/81-1740068');
+
+      done();
+    }, 150);
+  });
+
+  it('should donate throught paypal sucessfully', () => {
+    spyOn(siteAnalyticsServiceStub, 'registerGoToDonationSiteEvent')
+      .and.callThrough();
+
+    this.component.onDonateThroughPayPal();
+    expect(siteAnalyticsServiceStub.registerGoToDonationSiteEvent)
+      .toHaveBeenCalledWith('PayPal');
```

Comment:
Should you verify the URL here too? (Like the previous test)

Reply:
The function is not changing the location href on paypal.